spec/services/tax_tribs/rebuild_pdf_spec.rb

```diff
@@ -53,4 +53,36 @@
       described_class.rebuild
     end
   end
+
+  context "when everything is correct" do
+    context "it has the correct parameters for appeal" do
+      before do
+        @tc = TribunalCase.create(case_reference: 'TC/2016/12345', pdf_generation_status: 'APPEAL_ATTEMPT')
+        expect(TaxTribs::CaseDetailsPdf).to receive(:new).with(@tc, AppealCasesController, CheckAnswers::AppealAnswersPresenter)
```

Comment:
so the expect shouldn't be in before block. It should be in "it" block. same for the before block bellow. In before block you should use allow and then in it you will have "have_received"